app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEIViewModel.kt

```diff
@@ -51,17 +53,26 @@ class SearchTEIViewModel(
     val createButtonScrollVisibility = MutableLiveData(true)
     val isScrollingDown = MutableLiveData(false)
 
-    val allowCreateWithoutSearch = false // init from configuration
+    var allowCreateWithoutSearch = false
```

Comment:
Maybe instead of store this status in a property, you could get it any time via `searchRepository.canCreateInProgramWithoutSearch()`. with this we can store the data always in the same place.

Also instead of create a method only for testing like:
`@VisibleForTesting(otherwise = VisibleForTesting.NONE)
    fun setAllowCreateBeforeSearch(allowCreateWithoutSearch: Boolean) {
        this.allowCreateWithoutSearch = allowCreateWithoutSearch
    }`

  in the tests of `SearchTEIViewModelTest`, you can mock the repository result via ` whenever(searchRepository.canCreateInProgramWithoutSearch()) doReturn true or false`